packages/core/src/fhir-to-cda/cda-templates/components/family-history.ts

```diff
@@ -0,0 +1,308 @@
+      isFamilyMemberHistory(entry.resource) ? [entry.resource] : []
+    ) || [];
+
+  if (familyHistory.length === 0) {
+    return undefined;
+  }
+
+  const augmentedMemberHistories = familyHistory.map(memberHistory => {
+    return new AugmentedFamilyMemberHistory(familyHistorySectionName, memberHistory);
+  });
+
+  const { trs, entries } = createTableRowsAndEntries(
+    augmentedMemberHistories,
+    createTableRowsFromMemberHistory,
+    createEntryFromMemberHistory
+  );
+
+  const table = initiateSectionTable(familyHistorySectionName, tableHeaders, trs);
+
+  return {
+    templateId: buildInstanceIdentifier({
+      root: oids.familyHistorySection,
+      extension: extensionValue2015,
+    }),
+    code: buildCodeCe({
+      code: "10157-6",
+      codeSystem: loincCodeSystem,
+      codeSystemName: loincSystemName,
+      displayName: "Family History",
+    }),
+    title: "FAMILY HISTORY",
+    text: table,
+    entry: entries,
+  };
+}
+
+function createTableRowsFromMemberHistory(
+  augHistory: AugmentedFamilyMemberHistory,
+  referenceId: string
+): ObservationTableRow | ObservationTableRow[] {
+  const relationship = augHistory.resource.relationship;
+  const relationshipString = getDisplaysFromCodeableConcepts(relationship);
+  const name = augHistory.resource.name;
+
+  if (!augHistory.resource.condition) {
+    return createTableRowFromMemberHistory(undefined, relationshipString, name, referenceId);
+  }
+  return augHistory.resource.condition?.map((condition, index) => {
+    return createTableRowFromMemberHistory(
+      condition,
+      relationshipString,
+      name,
+      createConditionReference(referenceId, index)
+    );
+  });
+}
+
+function createConditionReference(ref: string, index: number): string {
+  return `${ref}-condition${index + 1}`;
+}
+
+function createTableRowFromMemberHistory(
+  condition: FamilyMemberHistoryCondition | undefined,
+  relationship: string | undefined,
+  name: string | undefined,
+  referenceId: string
+): ObservationTableRow {
+  return {
+    tr: {
+      _ID: referenceId,
+      ["td"]: [
+        {
+          "#text": getMedicalCondition(condition?.code) ?? NOT_SPECIFIED,
+        },
+        {
+          "#text": getConditionOnset(condition?.onsetAge) ?? NOT_SPECIFIED,
+        },
+        {
+          "#text": relationship ?? NOT_SPECIFIED,
+        },
+        {
+          "#text": name ?? NOT_SPECIFIED,
+        },
+        {
+          "#text": getNotes(condition?.note) ?? NOT_SPECIFIED,
+        },
+      ],
+    },
+  };
+}
+
+function getMedicalCondition(concept: CodeableConcept | undefined): string | undefined {
+  return concept ? getDisplaysFromCodeableConcepts(concept) : undefined;
+}
+
+function getConditionOnset(age: Age | undefined): string | undefined {
+  const ageString = `${age?.value ?? ""}${age?.unit ?? ""}`;
+  return ageString.length > 0 ? ageString : undefined;
+}
+
+function getNotes(note: Annotation[] | undefined): string | undefined {
```

Comment:
nit: common function right? not family member specific. maybe put in commons or wherever you deem appropriate to put shared code